--- CMakeLists.txt.release2.4.1	2018-03-19 19:04:34.000000000 +0100
+++ CMakeLists.txt	2019-03-03 12:31:20.348324794 +0100
@@ -13,6 +13,7 @@
 option( USE_TZ_DB_IN_DOT "Save the timezone database in the current folder" OFF )
 option( BUILD_SHARED_LIBS  "Build a shared version of library" OFF )
 option( ENABLE_DATE_TESTING "Enable unit tests" ON )
+option( DISABLE_STRING_VIEW "Disable string view" OFF )
 
 function( print_option OPT )
 	if ( NOT DEFINED PRINT_OPTION_CURR_${OPT} OR ( NOT PRINT_OPTION_CURR_${OPT} STREQUAL ${OPT} ) )
@@ -26,6 +27,7 @@
 print_option( USE_TZ_DB_IN_DOT )
 print_option( BUILD_SHARED_LIBS  )
 print_option( ENABLE_DATE_TESTING )
+print_option( DISABLE_STRING_VIEW )
 
 set( HEADER_FOLDER "include" )
 set( SOURCE_FOLDER "src" )
@@ -65,6 +67,15 @@
 	target_compile_definitions( tz PRIVATE -DINSTALL=. )
 endif( )
 
+if( DISABLE_STRING_VIEW )
+    target_compile_definitions( tz PRIVATE -DHAS_STRING_VIEW=0 )
+endif( )
+
+if( BUILD_SHARED_LIBS )
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fPIC")
+    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -fPIC")
+endif( )
+
 target_link_libraries( tz ${CMAKE_THREAD_LIBS_INIT} ${OPTIONAL_LIBRARIES} )
 
 # add include folders to the library and targets that consume it
